hazelcast/src/main/java/com/hazelcast/cache/impl/nearcache/impl/store/AbstractNearCacheRecordStore.java

```diff
@@ -270,6 +275,10 @@ public V get(K key) {
     public void put(K key, V value) {
         checkAvailable();
 
+        if (!isEvictionEnabled() && maxSizeChecker != null && maxSizeChecker.isReachedToMaxSize()) {
+            return;
```

Comment:
I don't think it is. The old implementation tries to expire values, but if the eviction policy is `NONE` and the Near Cache is full, it does nothing an just returns:
```
public void put(Data key, Object value) {
    fireCacheExpiration();
    if (evictionPolicy == NONE && cache.size() >= maxSize) {
        return;
    }
    if (evictionPolicy != NONE && cache.size() >= maxSize) {
        fireCacheEviction();
    }
    ...
}
```
The expiration is done regularly by the `ExpirationTask` in the `DefaultNearCache`, so we don't need to trigger it before checking the eviction policy.